### PR TITLE
Bump C++, CMake and update gitignore

### PR DESCRIPTION
**Pull request by sweetberryjuicee**
Just a little warning: Most of the C++ compilers doesn't support the new C++ 26 features.

### CODE REVIEW

**Comment by matcool** [comment]
> Just a little warning: Most of the C++ compilers doesn't support the new C++ 26 features.

then why bump it?? geode only requires c++20

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,19 +1,19 @@
-cmake_minimum_required(VERSION 3.21)
-set(CMAKE_CXX_STANDARD 20)
+cmake_minimum_required(VERSION 3.30)
```
Comment (matcool) [inline CMakeLists.txt]: this breaks compilation: https://discourse.cmake.org/t/cmake-3-28-cmake-cxx-compiler-clang-scan-deps-notfound-not-found/9244/3

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+set(CMAKE_CXX_STANDARD 26)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_OSX_ARCHITECTURES "arm64;x86_64")
 set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 
 project(Template VERSION 1.0.0)
 
-# Set up the mod binary
+# Set up the mod library.
```
Comment (matcool) [inline CMakeLists.txt]: no

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
 add_library(${PROJECT_NAME} SHARED
     src/main.cpp
-    # Add any extra C++ source files here
+    # Add any extra mod C++ files here.
```
Comment (matcool) [inline CMakeLists.txt]: y

File: CMakeLists.txt (hunk 1 above)
Comment (sweetberryjuicee) [reply CMakeLists.txt]: mb

File: README.md [hunk 4]
```diff
@@ -3,20 +3,20 @@ This is where she makes a mod.
...
+For more info, see [our docs](https://docs.geode-sdk.org/getting-started/create-mod/#build)
 ```sh
 # Assuming you have the CLI set up already
 geode build
 ```
 
 # Resources
 * [Geode SDK Documentation](https://docs.geode-sdk.org/)
-* [Geode SDK Source Code](https://github.com/geode-sdk/geode/)
+* [Geode SDK Source Code](https://github.com/geode-sdk/geode)
```
Comment (matcool) [inline README.md]: why

File: src/main.cpp [hunk 5]
```diff
@@ -4,7 +4,7 @@
 #include <Geode/Geode.hpp>
 
 /**
- * Brings cocos2d and all Geode namespaces to the current scope.
+ * Brings Cocos2d and all Geode namespaces to the current scope.
```
Comment (matcool) [inline src/main.cpp]: why

**Comment by sweetberryjuicee** [comment]
ok u can js make changes idk how to do the resolve thingy

**Comment by sweetberryjuicee** [comment]
then i will make my changes :p

File: CMakeLists.txt (hunk 1 above)
Comment (sweetberryjuicee) [reply CMakeLists.txt]: do i need to change cxx standard to 20 then?

File: README.md (hunk 4 above)
Comment (sweetberryjuicee) [reply README.md]: not needed lol

File: CMakeLists.txt (hunk 3 above)
Comment (sweetberryjuicee) [reply CMakeLists.txt]: yippie

File: CMakeLists.txt (hunk 2 above)
Comment (sweetberryjuicee) [reply CMakeLists.txt]: mb

File: src/main.cpp (hunk 5 above)
Comment (sweetberryjuicee) [reply src/main.cpp]: lol

**Comment by sweetberryjuicee** [comment]
ive done my last changes i think @matcool

**Comment by matcool** [comment]
> ive done my last changes i think @matcool

you still left the cmake version at 3.30 and c++ at 26, those should stay the same as they were, theres no reason to bump the version.

**Comment by sweetberryjuicee** [comment]
okay then reverting them

**Comment by sweetberryjuicee** [comment]
done

**Comment by sweetberryjuicee** [comment]
wait

**Comment by sweetberryjuicee** [comment]
ok added other files now

**Comment by sweetberryjuicee** [comment]
u can merge now @matcool

**Pull request merged**